### PR TITLE
apvs-385 - get auto approval config from database

### PR DESCRIPTION
**Pull request by pwright08**
apvs-385 - get auto approval config from database

## Checklist

- [x] Unit tests
- [ ] Code coverage checked
- [ ] Coding standards
- [ ] Error Handling
- [ ] Security/performance considered?
- [ ] Deployment changes considered?
- [ ] README updated

### CODE REVIEW

File: app/services/auto-approval/auto-approval-process.js [hunk 1]
```diff
@@ -1,69 +1,71 @@
...
+            var result = {checks: []}
 
-  if (autoApprovalEnabled) {
-    return getDataForAutoApprovalChecks(reference, eligibilityId, claimId)
-      .then(function (autoApprovalData) {
-        var result = {checks: []}
+            // Fail auto-approval check if First time claim, advance claim or status is PENDING
+            if (autoApprovalData.Claim.ClaimType === claimTypeEnum.FIRST_TIME ||
+              autoApprovalData.Claim.IsAdvanceClaim ||
+              autoApprovalData.Claim.Status === statusEnum.PENDING) {
```
Comment (stevenalexander) [inline app/services/auto-approval/auto-approval-process.js]: split into function that returns true/false if claim should be checked (removing need to comment)

File: app/services/auto-approval/auto-approval-process.js [hunk 2]
```diff
@@ -1,69 +1,71 @@
...
 
-        autoApprovalChecks.forEach(function (check) {
-          var checkResult = check(autoApprovalData)
-          result.checks.push(checkResult)
-        })
+            // Add AutoApprovalConfig
+            autoApprovalData.costVariancePercentage = config.CostVariancePercentage || '10'
+            autoApprovalData.maxNumberOfClaimsPerYear = config.MaxNumberOfClaimsPerYear || '26'
+            autoApprovalData.maxDaysAfterAPVUVisit = config.MaxDaysAfterAPVUVisit || '28'
+            autoApprovalData.maxClaimTotal = config.MaxClaimTotal || '250'
```
Comment (stevenalexander) [inline app/services/auto-approval/auto-approval-process.js]: as above split into function to add config to data

File: app/services/auto-approval/auto-approval-process.js [hunk 3]
```diff
@@ -1,69 +1,71 @@
...
-          if (!check.result) {
-            result.claimApproved = false
-          }
-        })
+            autoApprovalRulesEnum.forEach(function (check) {
+              if (disabledRules.indexOf(check) === -1) {
+                var checkResult = autoApprovalChecks[check](autoApprovalData)
+                result.checks.push(checkResult)
+              }
+            })
```
Comment (stevenalexander) [inline app/services/auto-approval/auto-approval-process.js]: as above split into function to get checks

File: app/services/auto-approval/auto-approval-process.js [hunk 4]
```diff
@@ -1,69 +1,71 @@
...
-        if (result.claimApproved) {
-          return autoApproveClaim(claimId, autoApprovalData.Visitor.EmailAddress)
-            .then(function () {
-              return result
+            result.claimApproved = true
+            // Loop through result properties, if any are false, then the claim should not be approved
+            result.checks.forEach(function (check) {
+              if (!check.result) {
+                result.claimApproved = false
+              }
```
Comment (stevenalexander) [inline app/services/auto-approval/auto-approval-process.js]: as above split into function to call checks and get result

**Comment by pwright08** [comment]
tests failing

File: app/services/auto-approval/auto-approval-process.js [hunk 5]
```diff
@@ -1,69 +1,80 @@
...
-    return Promise.resolve(null)
+      } else {
+        return Promise.resolve(null)
+      }
+    })
+}
+
+function failBasedOnPreRequisiteChecks (result, autoApprovalData) {
+  if (autoApprovalData.Claim.ClaimType === claimTypeEnum.FIRST_TIME ||
+    (autoApprovalData.Claim.Status === statusEnum.PENDING)) {
```
Comment (stevenalexander) [inline app/services/auto-approval/auto-approval-process.js]: `autoApprovalData.Claim.Status !== statusEnum.NEW`

**Pull request merged**